src/libraries/Common/src/System/Net/Security/SslKeyLogger.cs

```diff
@@ -0,0 +1,132 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+
+internal static class SslKeyLogger
+{
+    private static readonly string? s_keyLogFile = Environment.GetEnvironmentVariable("SSLKEYLOGFILE");
+    private static readonly FileStream? s_fileStream;
+
+#pragma warning disable CA1810 // Initialize all static fields when declared and remove cctor
+    static SslKeyLogger()
+    {
+        s_fileStream = null;
+
+        try
+        {
+#if DEBUG
+            bool isEnabled = true;
+#else
+            bool isEnabled = AppContext.TryGetSwitch("System.Net.Security.EnableSslKeyLogging", out bool enabled) && enabled;
```

Comment:
I'm wondering if we should add "Dangerous" to the name just to make it very clear.

Reply:
1. add Dangerous: seems redundant, we already have 2 levels you need to set up. Also no precedent for this with AppContext switches.
2. remove .Security: I'm up for it, but I don't mind the current state either (it's still a "security" feature).

Reply:
team consensus was to not add Dangerous or other signal words and remove .Security